src/stackwalk.c

```diff
@@ -34,12 +34,12 @@ static jl_gcframe_t *is_enter_interpreter_frame(jl_gcframe_t **ppgcstack, uintpt
         jl_gcframe_t *prev = pgcstack->prev;
         if (pgcstack->nroots & 2) { // tagged frame
             uintptr_t frame_fp = ((uintptr_t*)pgcstack)[-1];
-            if (frame_fp == 0)
-                continue; // frame wasn't fully initialized yet
-            if (frame_fp >= sp)
-                break; // stack grows down, so frame pointer is monotonically increasing
-            *ppgcstack = prev;
-            return pgcstack;
+            if (frame_fp != 0) { // check that frame was fully initialized
+                if (frame_fp >= sp)
+                    break; // stack grows down, so frame pointer is monotonically increasing
+                *ppgcstack = prev;
+                return pgcstack;
+            }
         }
         *ppgcstack = pgcstack = prev;
```

Comment:
I know it's not strictly related to this fix. But is there a need to set `*ppgcstack` here?

Reply:
Yes, this keeps track of which frames we've already looked at

Reply:
It's set before the return. Seemed kind of unnecessary to set it here.